controllers/front/return.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ *NOTICE OF LICENSE
+ *
+ *This source file is subject to the Open Software License (OSL 3.0)
+ *that is bundled with this package in the file LICENSE.txt.
+ *It is also available through the world-wide-web at this URL:
+ *http://opensource.org/licenses/osl-3.0.php
+ *If you did not receive a copy of the license and are unable to
+ *obtain it through the world-wide-web, please send an email
+ *to license@prestashop.com so we can send you a copy immediately.
+ *
+ *DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ *versions in the future. If you wish to customize PrestaShop for your
+ *needs please refer to http://www.prestashop.com for more information.
+ *
+ *@author INVERTUS UAB www.invertus.eu  <support@invertus.eu>
+ *@copyright SIX Payment Services
+ *@license   SIX Payment Services
+ */
+
+use Invertus\SaferPay\Config\SaferPayConfig;
+use Invertus\SaferPay\Controller\AbstractSaferPayController;
+use Invertus\SaferPay\DTO\Response\Assert\AssertBody;
+use Invertus\SaferPay\Service\TransactionFlow\SaferPayTransactionAssertion;
+
+class SaferPayOfficialReturnModuleFrontController extends AbstractSaferPayController
+{
+    const FILENAME = 'return';
+
+    /**
+     * @throws PrestaShopException
+     */
+    public function postProcess()
+    {
+        $cartId = Tools::getValue('cartId');
```

Comment:
You added these params, but i don't see it being passed anywhere, i'm not sure if orderId would exist when creating payment, so would need to use only cartId.